airflow/www/static/js/api/useTaskDetail.tsx

```diff
@@ -0,0 +1,34 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios, { AxiosResponse } from "axios";
+import type { API } from "src/types";
+import { useQuery } from "react-query";
+import { getMetaValue } from "../utils";
+
+const taskDetailURI = getMetaValue("task_detail_api");
+
+export default function useTaskDetail({ taskId }: { taskId: string }) {
+  return useQuery(["taskDetails", taskId], async () => {
+    const url = taskDetailURI.replace("_TASK_ID_", taskId);
+
+    const datum = await axios.get<AxiosResponse, API.Task>(url);
+    return datum;
```

Comment:
Small nit, does it make sense to define a const just to return the same? Can be a one-liner
```suggestion
    return await axios.get<AxiosResponse, API.Task>(url);
```